frontend/src/context/AuthContext.jsx

```
@@ -0,0 +1,109 @@
+    setLoading(true);
+    try {
+      const data = await fetchDataFromEndpoint(
+        endpoint,
+        "POST",
+        { email, password },
+        false // Login does not require authentication initially
+      );
+
+      // Assuming the backend returns the authenticated entity in `data.entity`
+      // and the token in `data.token`
+      if (data && data.entity && data.token) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)